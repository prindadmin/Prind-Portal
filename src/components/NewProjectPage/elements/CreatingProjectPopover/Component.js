import React, { Component } from 'react'
//import PropTypes from 'prop-types'

import PopOverHandler from '../../../common/popOverHandler'

import {
  Spinner,
  Intent,
} from '@blueprintjs/core'

import * as strings from '../../../../data/Strings'

export class Element extends Component {
  static propTypes = {
  }

  getContent = () => {
    return (
      <React.Fragment>
        {strings.PROJECT_CREATION_BILLING_WARNING}
        <Spinner size='150' intent={Intent.DANGER}/>
      </React.Fragment>
    )
  }

  render () {
    return (
      <PopOverHandler>
        <div id='popup-greyer' onClick={(e) => e.stopPropagation()}>
          <div id='project-creation-popover'>
            <div id='popup-box'>
              <div className='project-creation-popover-container' onClick={(e) => e.stopPropagation()}>
                <div className='element-title'>
                  {strings.PROJECT_CREATION_IN_PROGRESS}
                </div>
                <div className='element-description'>
                  {this.getContent()}
                </div>
              </div>
            </div>
          </div>
        </div>
      </PopOverHandler>
    )
  }

}

export default Element
