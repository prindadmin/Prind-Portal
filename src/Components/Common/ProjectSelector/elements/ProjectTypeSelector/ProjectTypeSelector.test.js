import React from 'react';
import { shallow } from 'enzyme';

import Component from './ProjectTypeSelector'

it('Should render', () => {
  const props = {
    closePopup: function() {},
    history: {
      push: function() {},
    },
  };
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});


it('Should create new project', () => {
  const mockClosePopup = jest.fn()
  const mockHistoryPush = jest.fn()
  const props = {
    closePopup: mockClosePopup,
    history: {
      push: mockHistoryPush,
    },
  };
  const component = shallow(<Component {...props} />);
  component.find('#DHSFProject').simulate('select', 'DHSFProject')
  expect(mockHistoryPush).toHaveBeenCalled()
  expect(mockClosePopup).toHaveBeenCalled()
});
