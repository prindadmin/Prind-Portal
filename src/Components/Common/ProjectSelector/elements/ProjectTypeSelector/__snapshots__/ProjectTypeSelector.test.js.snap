// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render 1`] = `
<div
  className="project-selector-container"
>
  <lazy
    description="A new project that is in the inception stage.  This type of project will guide you from inception to occupation through the construction process."
    icon="building"
    id="CDM2015Project"
    key="0"
    name="CDM2015 Project"
    onSelect={[Function]}
  />
  <lazy
    description="An existing build that you wish to store the Health & Safety File for.  This type of project will allow you to store all required parts of the Health & Safety File digitally."
    icon="building"
    id="DHSFProject"
    key="1"
    name="DHSF Project"
    onSelect={[Function]}
  />
</div>
`;
